h5_test/tst_h_dimscales.c

```diff
@@ -379,7 +379,7 @@ main()
       if ((grpid = H5Gopen(fileid, GRP_NAME)) < 0) ERR;
 
       /* Loop through datasets to find variables. */
-      if (H5Gget_num_objs(grpid, &num_obj) < 0) ERR;
+      if (H5Gget_num_objs(grpid, (hsize_t *)&num_obj) < 0) ERR;
```

Comment:
Made this change to eliminate warning. H5Gget_num_objs does not expect a volatile.